### PR TITLE
Expand video embeds to fullwidth

### LINKED ISSUE #926: Improved video embeds for the browser
### Is your proposal related to a problem?

At the moment, videos will be embedded as 300 px x 150 px iframes with the fullscreen option disabled. This provides a sub-optimal experience for watching the embedded videos through the browser.

image

### Describe the solution you'd like

My proposed changes are:
- Allow fullscreen
- Center the frame
- Make the frame larger

I have come up with a solution that works for me. In this solution, I have hard-coded the width and height to 560 px by 315 px (19:6), centered the frame with <center></center>, removed the frame border, and allowed full-screen. Here is an example: https://mander.xyz/post/533860

image

For this solution, the [<iframe> tag](https://github.com/LemmyNet/lemmy-ui/blob/4eb902306df7f4751cb5b7f4caa6aba48a8ddc5a/src/shared/components/post/metadata-card.tsx#L78) is modified to:

`<center><iframe allowFullscreen="allowfullscreen" src={post.embed_video_url} width="560px" height="315px" frameborder="0"></iframe></center>`

### Describe alternatives you've considered

I have also tested what this does to the display on a phone screen. Unfortunately, setting the absolute height and width will cause the video to appear large in phone screens. I tried to find a simple way to create a responsive iframe, but the solutions that I have found require modifying the CSS code or running some javascript code to resize. The solution I propose is the simplest reasonable solution I could come up with. 

### Additional context

I will submit a PR in case you choose to adopt this approach. At the very least, I would suggest allowing the videos to be full-screened.

### PR DESCRIPTION
**Pull request by cloventt**
Use Bootstrap 5.0 `.ratio` class to wrap embedded video iframe so that it is properly handled in a responsive way. Also add `allowfullscreen` hint and `title` attribute to embedded iframe.

Fixes #926

### CODE REVIEW

**Comment by cloventt** [comment]
TODO: the text `Embedded Video` should probably be handled as a translatable but I wasn't sure how the translation repo integrates with this one exactly.

**Comment by cloventt** [comment]
I haven't been able to test this locally as I haven't been able to get a post with an embed to work properly for some reason.

**Comment by cloventt** [comment]
Tested and can confirm it looks great: https://imgur.com/a/ho0nUIh

The video can also be made fullscreen.

How it renders on desktop

How it renders on mobile

File: src/shared/components/post/metadata-card.tsx [hunk 1]
```diff
@@ -75,10 +75,14 @@ export class MetadataCard extends Component<
...
-          <iframe
-            className="post-metadata-iframe"
-            src={post.embed_video_url}
-          ></iframe>
+          <div className="ratio ratio-16x9">
+            <iframe
+              allowFullScreen
+              className="post-metadata-iframe"
+              src={post.embed_video_url}
+              title={"Embedded Video: " + post.embed_title}
```
Comment (SleeplessOne1917) [inline src/shared/components/post/metadata-card.tsx:83]: If you add the "Embedded Video" string, there will have to be a [translation made.](https://github.com/LemmyNet/lemmy-translations). Alternatively, you could just make the title `post.embed_title`.

Comment (cloventt) [reply src/shared/components/post/metadata-card.tsx:83]: Ok, for now I will remove the prefix, but it is probably worth adding in the future as screen readers might find it useful.

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**